sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -613,7 +613,10 @@ class ParquetFilters(
     value == null || (nameToParquetField(name).fieldType match {
       case ParquetBooleanType => value.isInstanceOf[JBoolean]
       case ParquetIntegerType if value.isInstanceOf[Period] => true
-      case ParquetByteType | ParquetShortType | ParquetIntegerType => value.isInstanceOf[Number]
+      case ParquetByteType | ParquetShortType | ParquetIntegerType => value match {
```

Comment:
I think you should add some comments like there are only INT32 and INT64 physical types in Parquet format specification; therefore, we don't check INT8 and/or INT16.

Reply:
Or, we could even allow exact match of `value` on `Int` to be conservative? cc @cloud-fan and @wangyum

Reply:
Added a comment. I also made the condition more restrictive: value must be either a java `Byte`/`Short`/`Integer`/`Long`. This excludes other `Number`s such as `Float`/`Double`/`BigDecimal` that are generally not safe to cast to int and that the initial change didn't block. For example, float `NaN` gets cast to `0` and would pass the previous check.

We can't match on `Integer` because for `ParquetByteType` and `ParquetShortType` the value will typically be a `Byte` and `Short` resp. Also this allows upcasting where a parquet integer type is read with a larger Spark integer type.